games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+            selectedCards.forEach((c) => {
+              c.clearTint();
+              c.disableInteractive();
+              this.disableCardDraggable();
+              player.removeCardFromHand(c);
+            });
+            this.arrayCardsPutOnMeldZone.push(
+              selectedCards
+            );
+          });
+        }
+      };
+
+      player.hand.forEach((card) => {
+        card.on('pointerdown', () => {
+          handleCardSelection(card);
+        });
+      });
+
+      await new Promise<void>((resolve) => {
+        const checkSelection = () => {
+          if (selectedCards.length === 3) {
+            // 手札を更新
+            this.resetHandsMove(player);
+            console.log('処理終わり');
+            selectedCards = [];
+            resolve();
+          } else {
+            console.log(checkSelection);
+            setTimeout(checkSelection, 100);
+          }
+        };
+        checkSelection();
+      });
+    }
+  }
+
+  private enableCardDraggable(): void {
+    this.players.forEach((player) => {
+      if (player.playerType === 'player') {
+        player.hand.forEach((card) =>
+          card.setInteractive()
+        );
+      }
+    });
+  }
+
+  private disableCardDraggable(): void {
+    this.players.forEach((player) => {
+      if (player.playerType === 'player') {
+        player.hand.forEach((card) =>
+          card.disableInteractive()
+        );
+      }
+    });
+  }
+
+  private enableDeckFrontDraggable() {
+    this.frontDeckZone?.setInteractive();
+  }
+
+  private enbleDeckBacDraggable() {
+    this.backDeckZone?.setInteractive();
+  }
+
+  private async createDropZoneEvent() {
+    return new Promise<void>((resolve) => {
+      this.createCardDragStartEvent();
+      this.createCardDragEvent();
+      this.createCardDropEvent();
+
+      const handleDragEnd = (
+        pointer: Phaser.Input.Pointer,
+        card: Card,
+        dropped: boolean
+      ) => {
+        if (!dropped) {
+          const droppedCard = card;
+          droppedCard.x = droppedCard.input.dragStartX;
+          droppedCard.y = droppedCard.input.dragStartY;
+          this.children.bringToTop(droppedCard);
+          droppedCard.playMoveTween(
+            this.deckFrontCardX,
+            this.deckFrontCardY
+          );
+          this.players[0].removeCardFromHand(droppedCard);
+          this.deckFront.push(droppedCard);
+        }
+
+        this.input.off('dragend', handleDragEnd); // ハンドラを解除
+        resolve();
+      };
+
+      this.input.on('dragend', handleDragEnd);
+    });
+  }
+
+  private async createCardDropEvent() {
+    this.input.on(
+      'drop',
+      (
+        pointer: Phaser.Input.Pointer,
+        card: Card,
+        dropZone: Zone
+      ) => {
+        const droppedCard = card;
+        droppedCard.x = dropZone.x;
+        droppedCard.y = dropZone.y;
+
+        droppedCard.input.enabled = false;
+      }
+    );
+  }
+
+  private addPutOnHand(player: RummyPlayer) {
```

Comment:
関数名が意味わからないので、dealCardとかにしたらどうですか？


```suggestion
  private dealCard(player: RummyPlayer) {
```